src/preact/base-element.js

```diff
@@ -1083,9 +1087,7 @@ function parsePropDefs(Ctor, props, propDefs, element, mediaQueryProps) {
     // as separate properties. Thus in a carousel the plain "children" are
     // slides, and the "arrowNext" children are passed via a "arrowNext"
     // property.
-    const nodes = element.getRealChildNodes
-      ? element.getRealChildNodes()
-      : toArray(element.childNodes);
+    const nodes = realChildNodes(element);
```

Comment:
Would this change handle a basic `Element` less efficiently since it's not using the built-in iterator here?

Reply:
Thats a good question! I don't know why the underlying `realChildNodes` isn't directly using `element.childNodes`, since it seems to be attempting to do the same thing.  @jridgewell do you know?

Reply:
It looks like this is checking whether `getRealNodes` is defined, ie "Is this a CustomElement? If so, use our helper to get the real children, otherwise it's an ordinary element and we should use its children because they're all "real""

Reply:
Definitely what it is doing. My question is: "why does `childNodes` exist as a helper".
https://github.com/ampproject/amphtml/blob/c88b27558d9b139b29b22324692b5673f60cd629/src/core/dom/query.js#L250-L265

Can we replace it with `toArray(node.childNodes)`

Reply:
From further investigation, this is only used in two places. Will remove it entirely in a followup PR.